harvard-lts/drs-import-management-service
# Add basic publishing messaging capabilities

**Add basic publishing messaging capabilities**
* * *

**GitHub Issue**: [(link)](https://app.zenhub.com/workspaces/hdc-62163f60fb08c60011fa4795/issues/harvard-lts/hdc/18)

# What does this Pull Request do?
This PR includes the following relevant changes:

- Basic message publisher, which publishes an empty dummy message to a MQ.
- OpenAPI spec file for API endpoint documentation and OpenAPI-Controllers mapping using connexion.
- Local integration test environment.
- More advanced code layering, separating domain and application layers at the moment (Exception handling and propagation is also layer-oriented).
- Dependency Injection setup.

_/ingest endpoint currently does not receive a request body and returns a basic response for a first pass. Both the name of the endpoint and the request and response bodies are open to modification when the DIMS endpoints are finally defined._

# How should this be tested?

A description of what steps someone could take to:
* Create/Update .env and .test.env files as described in README.md
* Run the local development environment: `docker-compose -f docker-compose-dev.yml up`
* Access local development ActiveMQ frontend http://localhost:8161/admin/ (User: admin, Password: admin)
* Go to queues section: http://localhost:8161/admin/queues.jsp
* POST to http://localhost:10580/ingest
* In the queues section, there should appear a message posted to dev_queue
* Run the local test environment: `docker-compose -f docker-compose-test.yml up`
* Execute integration test TestInitiateIngestQueuePublisher: `docker exec -it test_runner pytest test/integration`

# Test coverage
- unit tests Yes
- integration tests Yes

# Interested parties
@jessjass @ives1227
